rotkehlchen/chain/ethereum/modules/compound/v2/decoder.py

```diff
@@ -176,16 +181,27 @@ def _decode_borrow_and_repay(
         amount = asset_normalized_value(amount_raw, underlying_asset)
         for event in decoded_events:
             # Find the transfer event which should have come before the redeeming
-            if event.event_type == HistoryEventType.RECEIVE and event.asset.identifier == underlying_asset and event.balance.amount == amount:  # noqa: E501
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.asset.identifier == underlying_asset and
+                event.balance.amount == amount and
+                event.address == compound_token.evm_address
```

Comment:
@LefterisJP : for my previous comment about "consistency", that started from here because I needed to add another "and" condition to this line that was already long and already suppressing the E501. If you prefer that I revert the other sections where I changed only the indendation, what do you prefer for this line? Extending the one line if, or kepping in this way as I committed?